js/apps/system/_admin/aardvark/APP/frontend/js/routers/startApp.js

```diff
@@ -13,6 +13,27 @@
       }
     });
 
+    const updateOptions = function(options) {
+      const update = { ...options };
```

Comment:
@adityamukho I disagree here, I would recommend *not* modifying function parameters at all, and always making copies. This will ensure immutability across the codebase. The cost is minimal, and it avoids a plethora of potential bugs.

Modifying incoming function params is a bad idea in general, I think we should even add a lint rule against it

Reply:
Even the function name is okay, but if needed, we can rename the function to `getUpdatedOptions`. 

We should avoid mutation. It causes bad patterns across the codebase, and in React-land it's potentially fatal.

Reply:
Good points. I do have a few comments:

1. By and large I'm OK with copying the input object, esp since in this case it is a small one. However, we should also observe that shallow copies do not actually ensure immutability.
2. Input-altering functions are actually quite a commonly used pattern in several languages, including JS. As with everything, there are places where they're a good fit, and those where they're not. This, IMO is a case of the former.
3. Immutability, like any other pattern, is not meant to be dogmatically applied everywhere.
4.  This is not yet a React app. Even inside a React app, there are places where immutables are preferred and for well understood reasons (e.g. state, props, globals and any other objects shared across several components), but we still don't need to enforce immutability for small, nearly inline helper functions that only work with objects whose scopes and lifetimes are tightly constrained.

Reply:
Sure, fair enough, but as you said, this is a small object, so it's okay to make a copy. Let's keep this the way it is as of now. My motivation is ease of quickly scanning a function and figuring out what it's doing. Immutability helps.